### DISCUSSION TITLE
Code Artifacts

### DISCUSSION THREAD

**Discussion by reramjiawan** [danny-avila/LibreChat#3970] [Troubleshooting]
Hi, from my understanding Code Artifacts when enabled through the settings in the Beta features tab should work out of the box for all models. However its not working for me. Does it only work for Anthropic? IDK what I am missing here or doing wrong. 

When checking the Presentation Component (Presentation.tsx) the code that renders the Artifacts seems to be null always so it never renders.
```
Object.keys(artifacts ?? {}).length > 0 ? (
  <Artifacts />
) : null
```

**Comment by reramjiawan** [comment]
I have looked further into the backend and I noticed that for the build.js script, artifacts is undefined. So no artifact prompt can be generated. 

```
if (typeof artifacts === 'string') {
    endpointOption.artifactsPrompt = generateArtifactsPrompt({ endpoint, artifacts });
}
```

Chat post endpoint
```
{
    "text": "try again using code artifacts",
    "sender": "User",
    "isCreatedByUser": true,
    "parentMessageId": "e51d1871-c2c3-458d-99c7-1c92e9a1c634",
    "conversationId": "3f450e67-42ef-4388-adf6-f8b9d5882b32",
    "messageId": "894a228a-e4d7-4a5c-bdc5-2b993f4703ea",
    "error": false,
    "generation": "",
    "responseMessageId": "e51d1871-c2c3-458d-99c7-1c92e9a1c634",
    "overrideParentMessageId": null,
    "endpoint": "bedrock",
    "artifacts": "default",
    "region": "eu-central-1",
    "model": "anthropic.claude-3-5-sonnet-20240620-v1:0",
    "key": "never",
    "isContinued": false
}
```

Artifacts is set to default but the backend does not seem to receive it.


@danny-avila Appreciate your input here.

**Reply by reramjiawan** [reply]
So for Bedrock I have narrowed it down that the Zod schema for bedrock is missing `artifacts: true`. However even with this addition still no artifacts in the frontend 😞 .

**Reply by reramjiawan** [reply]
Works for Azure Open AI:

![image](https://github.com/user-attachments/assets/be8693c9-44c6-462f-99a6-ad7bcd6c7df5)

**Reply by danny-avila** [reply]
Hi, artifacts is not implemented for bedrock yet. I will get to it today.

**Reply by danny-avila** [reply]
Merging here: https://github.com/danny-avila/LibreChat/pull/3974

**Reply by AxelFooley** [reply]
> Hi, artifacts is not implemented for bedrock yet. I will get to it today.

Can we have a list of where artifacts are implemented? i am using a local Ollama instance and managed to get it work by chance only once, at this point i assume that it depends on the model i am using? Sadly i cannot remember which one i was using that one time it worked.

**Reply by danny-avila** [reply]
It all comes down to prompting. Unfortunately, for most open source models, the current required format for artifacts is a bit too "advanced" for consistent output, which is why the "smarter" models, i.e. claude, gpt-4, etc, will perform more consistently

**Reply by AIndoria** [reply]
Can I ask a related question? When I do artifacts on OpenAI, it doesn't do the <1/1, 2/2> increment bit on the artifacts UI, and  keeps getting stuck on 1/1 forever etc. Anthropic seems to work fine (anthropic ironically also has low daily limits because of low tier ). Is that intended? I am assuming some sort of bug.
Further, forking long conversations with artifacts (even with claude) seems a bit borked right now. It only seems to include the conversation up to the first artifact that it displays upon refresh.
...now that I think about it I should probably open a new issue. 😅

**Reply by danny-avila** [reply]
I can’t reproduce either issue, try making it a discussions thread first @AIndoria
